DigitalLearningSolutions.Data/DataServices/ProgressDataService.cs

```diff
@@ -378,5 +399,70 @@ public IEnumerable<DetailedTutorialProgress> GetTutorialProgressDataForSection(i
                 new { progressId, promptNumber, answer }
             );
         }
+
+        public void UpdateProgressDetailsForStoreAspProgressV2(
+            int progressId,
+            int customisationVersion,
+            DateTime submittedTime,
+            string progressText
+        )
+        {
+            connection.Execute(
+                @"UPDATE Progress
+                    SET CustomisationVersion = @customisationVersion,
+                        SubmittedTime = @submittedTime,
+                        ProgressText = @progressText,
+                        DiagnosticScore =
+                            (SELECT CASE WHEN SUM(t.DiagAssessOutOf) > 0
+                                THEN CAST((SUM(ap.DiagHigh) * 1.0) / (SUM(t.DiagAssessOutOf) * 1.0) * 100 AS Int)
+                                ELSE 0 END AS DiagPercent
+                            FROM aspProgress AS ap
+                            INNER JOIN Progress AS p ON ap.ProgressID = p.ProgressID
+                            INNER JOIN Customisations AS c ON p.CustomisationID = c.CustomisationID
+                            INNER JOIN CustomisationTutorials AS ct ON ap.TutorialID = ct.TutorialID AND c.CustomisationID = ct.CustomisationID
+                            INNER JOIN Tutorials AS t ON ct.TutorialID = t.TutorialID
+                            WHERE (ap.ProgressID = @progressId) AND (ct.DiagStatus = 1))",

```

Comment:
Looks the final `WHERE (ProgressID = @progressId)` that was under this line has been removed, I suspect by accident. If I am reading correctly, this will now update every single progress record in existence.